core/src/main/scala/io/projectglow/vcf/VCFLineToInternalRowConverter.scala

```diff
@@ -0,0 +1,532 @@
+/*
+ * Copyright 2019 The Glow Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.projectglow.vcf
+
+import java.util
+import java.util.Collections
+
+import com.google.common.base.Splitter
+import htsjdk.samtools.ValidationStringency
+import htsjdk.samtools.util.OverlapDetector
+import htsjdk.variant.vcf.VCFHeader
+import org.apache.hadoop.io.Text
+import org.apache.spark.sql.SQLUtils
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.expressions.GenericInternalRow
+import org.apache.spark.sql.catalyst.util.GenericArrayData
+import org.apache.spark.sql.types.{ArrayType, BooleanType, DataType, DoubleType, IntegerType, LongType, StringType, StructField, StructType}
+import org.apache.spark.unsafe.types.UTF8String
+import org.apache.spark.unsafe.types.UTF8String.{IntWrapper, LongWrapper}
+import io.projectglow.common.{GenotypeFields, HasStringency, SimpleInterval, VariantSchemas}
+
+import scala.util.control.NonFatal
+
+/**
+ * Converts the raw bytes in a VCF line into an [[InternalRow]]
+ *
+ * @param header The VCF header object this is currently only used to extract sample ids
+ * @param schema The schema of the converted rows
+ * @param stringency How to handle errors
+ * @param overlapDetectorOpt If provided, the converter will check to see if a variant passes this detector before
+ *                           parsing genotypes
+ */
+class VCFLineToInternalRowConverter(
```

Comment:
If I have not missed it, it seems validations against the `Number` for each field in the header is lost in this reader. I would assume `VariantContext` should have such validation. For example, if `Number = R` for a genotype fied do we check that is actually the case and perhaps warn or even create nulls in the array?

Reply:
Never mind this. I checked and it is coded in VariantContext but is not used.